src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1069,907 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+            if not (configmap_rg_name.lower() == resource_group_name.lower() and
+                    configmap_cluster_name.lower() == cluster_name.lower()):
+                telemetry.set_user_fault()
+                telemetry.set_exception(exception='The provided cluster name and rg correspond to different cluster', fault_type=consts.Operate_RG_Cluster_Name_Conflict,
+                                        summary='The provided cluster name and resource group name do not correspond to the kubernetes cluster being operated on.')
+                raise CLIError("The provided cluster name and resource group name do not correspond to the kubernetes cluster you are operating on. " +
+                               "Please use the cluster, with correct resource group and cluster name.")
+        else:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception='The corresponding CC resource does not exist', fault_type=consts.Corresponding_CC_Resource_Deleted_Fault,
+                                    summary='CC resource corresponding to this cluster has been deleted by the customer')
+            raise CLIError("There exist no ConnectedCluster resource corresponding to this kubernetes Cluster." +
+                           "Please cleanup the helm release first using 'az connectedk8s delete -n <connected-cluster-name> -g <resource-group-name>' and re-onboard the cluster using " +
+                           "'az connectedk8s connect -n <connected-cluster-name> -g <resource-group-name>'")
+
+    else:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception="The azure-arc release namespace couldn't be retrieved", fault_type=consts.Release_Namespace_Not_Found,
+                                summary="The azure-arc release namespace couldn't be retrieved, which implies that the kubernetes cluster has not been onboarded to azure-arc.")
+        raise CLIError("The azure-arc release namespace couldn't be retrieved, which implies that the kubernetes cluster has not been onboarded to azure-arc." +
+                       "Please run 'az connectedk8s connect -n <connected-cluster-name> -g <resource-group-name>' to onboard the cluster")
+    return release_namespace
+
+
+def get_all_helm_values(release_namespace, kube_config, kube_context):
+    cmd_helm_values = ["helm", "get", "values", "--all", "azure-arc", "--namespace", release_namespace]
+    if kube_config:
+        cmd_helm_values.extend(["--kubeconfig", kube_config])
+    if kube_context:
+        cmd_helm_values.extend(["--kube-context", kube_context])
+
+    response_helm_values_get = Popen(cmd_helm_values, stdout=PIPE, stderr=PIPE)
+    output_helm_values, error_helm_get_values = response_helm_values_get.communicate()
+    if response_helm_values_get.returncode != 0:
+        if ('forbidden' in error_helm_get_values.decode("ascii") or 'timed out waiting for the condition' in error_helm_get_values.decode("ascii")):
```

Comment:
'timed out waiting for the condition' in error_helm_get_values.decode("ascii"))
This checkk is only needed for helm install/upgrade

Reply:
Yepp, removing